EjercicioClases/Controller/DatabaseController.pm

```diff
@@ -12,52 +12,36 @@ use Model::TransactionRegistry;
 sub new{
     my $class = shift;
     bless {}, $class;
-    #return $self;
+    return $class;
 }
 
-my @sellers = ();
-my @buyers = ();
-my @products = ();
-my @transactions = ();
-
-sub saveSeller{
-    my ($class, $seller) = @_;
-    push(@sellers, $seller);
-}
-
-sub saveBuyer{
-    my ($class, $buyer) = @_;
-    push(@buyers, $buyer);
-}
-
-sub saveProduct{
-    my ($class, $product) = @_; #Forma indicada
-    push (@products, $product);
-}
-
-sub saveTransaction{
-    my ($class, $transaction) = @_;
-    push (@transactions, $transaction);
-}
+my %tables = (
+	sellers => [],
+	buyers => [],
+	products => [],
+    transactions => [],
+);
 
 sub save{
-    my ($tablename, $transaction) = @_;
+    my ($class, $tablename, $object) = @_;
+    
+    foreach my $key (keys %tables) {
```

Comment:
You don't need to iterate over the complete 'tables' hash. You can just evaluate if the tables hash contains the indicated 'tablename' and do the push of your new object.